### DISCUSSION TITLE
本地视频预览图，对比alist与ftp

### DISCUSSION THREAD

**Discussion by KnIfER** [AlistGo/alist#5828] [Ideas]
本地视频预览图，相信不少人感兴趣~
可以用ffmpeg获取。我的建议是另开服务进程（外挂）。

我用java开启的ftp服务，魔改协议，可实时生成缩略图（支持解析lnk连接，无需多开ftp账户）。配合glide，速度还不错！

至于为何多此一举？这就涉及ftp与alist的"本机存储"的区别：
- 同一播放器， alist 似乎比ftp快一些，尤其信号不好的情况下。
- alist 提供的http连接支持更多播放器。ftp只有部分版本的ffmpeg支持，比如ijkplayer，exo、vlc、安卓原生可能并不支持。

加之，缩略图的生成不能卡主或者影响主要进程，所以最好分开，外挂式加入这一功能。

目前是硬编码，写死了，只能自己用……

![image](https://github.com/alist-org/alist/assets/11593903/7262e158-52e5-497b-a0a8-face8bb79522)

**Comment by KnIfER** [comment]
原来是没人感兴趣吗，呵呵

![image](https://github.com/alist-org/alist/assets/11593903/6b0c1009-6390-48c4-a89b-e245ccb75090)

碍某人眼了。

**Reply by yeyudoc** [reply]
怎么做啊，是每次重新读取缩略图吗

**Comment by IsKenKenYa** [comment]
所以我们应该如何使用ffmpeg添加视频封面？

**Comment by gdtv** [comment]
这是个好思路，我也是先在本地用ffmpeg生成缩略图，再将缩略图和视频一起传到alist，不过我不会写alist插件，感谢你的提示的插件示例代码。

**Reply by yeyudoc** [reply]
怎么做啊，能简单说一下吗
